yaml/com/ibm/VPD/FRUManager.interface.yaml

```diff
@@ -0,0 +1,139 @@
+      returns:
+          - name: bytesUpdated
+            type: ssize
+            description: >
+                On success, returns the number of bytes updated. On failure,
+                returns -1.
+      errors:
+          - xyz.openbmc_project.Common.Error.InvalidArgument
+          - xyz.openbmc_project.Common.Error.InternalFailure
+
+    - name: ReadKeyword
+      description: >
+          Method to read a single keyword's value based on the input parameters.
+      parameters:
+          - name: path
+            type: string
+            description: >
+                Path where the data resides(D-Bus inventory path/EEPROM path
+                etc).
+          - name: paramsToReadData
+            type: variant[struct[string,string],struct[string]]
+            description: >
+                Parameters in required format to read the keyword data.
+      returns:
+          - name: keywordValue
+            type: variant[array[byte]]
+            description: >
+                On success, returns the Keyword's value read from the given
+                path. On failure, throws an exception.
+      errors:
+          - xyz.openbmc_project.Common.Error.InvalidArgument
+          - xyz.openbmc_project.Common.Error.InternalFailure
+
+    - name: GetLocationCode
+      description: >
+          Method to get location code from the D-Bus inventory path.
+      parameters:
+          - name: inventoryPath
+            type: object_path
+            description: >
+                D-Bus inventory path of the FRU.
+      returns:
+          - name: locationCode
+            type: string
+            description: >
+                On success, returns location code. On failure, throws an
+                exception.
+      errors:
+          - xyz.openbmc_project.Common.Error.InvalidArgument
+          - xyz.openbmc_project.Common.Error.InternalFailure
+
+    - name: GetExpandedLocationCode
+      description: >
+          Method to get expanded location code for a given un-expanded location
+          code.
+      parameters:
+          - name: locationCode
+            type: string
+            description: >
+                Location code in un-expanded format.
+          - name: nodeNumber
+            type: uint16
+            description: >
+                Denotes the node in case of multi-node configuration. Defaulted
+                in case of single node configuration.
+      returns:
+          - name: locationCode
+            type: string
+            description: >
+                Location code in expanded format.
+      errors:
+          - xyz.openbmc_project.Common.Error.InvalidArgument
+          - xyz.openbmc_project.Common.Error.InternalFailure
+
+    - name: deleteFRUVPD
+      description: >
+          Method to remove the FRU VPD.
+      parameters:
+          - name: inventoryPath
+            type: object_path
+            description: >
+                D-Bus inventory path of the FRU.
+      errors:
```

Comment:
is this how we define functions which throw and have no return variables ?

Reply:
yes

Reply:
I am ok with it